src/librustc_borrowck/diagnostics.rs

```diff
@@ -286,6 +286,66 @@ You can read more about cell types in the API documentation:
 https://doc.rust-lang.org/std/cell/
 "##,
 
+E0389: r##"
+An attempt was made to mutate data using a non-mutable reference. This
+commonly occurs when attempting to assign to a non-mutable reference of a
+mutable reference (`&(&mut T)`).
+
+Example of erroneous code:
+
+```compile_fail
+struct FancyNum {
+    num: u8
+}
+
+fn main() {
+    let mut fancy = FancyNum{ num: 5 };
+    let fancy_ref = &(&mut fancy);
+    fancy_ref.num = 6; // error: cannot assign to data in a `&` reference
+    println!("{}", fancy_ref.num);
+}
+```
+
+Here, `&mut fancy` is mutable, but `&(&mut fancy)` is not. To fix this, either
```

Comment:
Could you provide a more in-depth explanation as to why &&mut is immutable? This can trip up newcomers who come across it, since it looks like the inner value should be mutable.

(It isn't because &&mut can be aliased, which means that &T transitively makes all mutable references inside it immutable)